readthedocs/misc/v2-migration-guide.rst

```diff
@@ -202,6 +202,12 @@ is decided by Telegram.
 The ``telethon.errors`` module continues to provide custom errors used by the library such as
 ``TypeNotFoundError``.
 
+Errors are all now immutable - you can't modify them. This means the library will not correct
+weird errors such as FLOOD_WAIT_0 (which was previously modified to FLOOD_WAIT_1). When the
+library does automatic flood sleeping, it will still make this correction, but if the
+``flood_sleep_threshold`` is in the range [0,1) then you may get FLOOD_WAIT_0 raised to your
```

Comment:
This seems a bit strange. What is the rationale for wanting to have flood 0 errors be propagated even when the threshold is 0?

Reply:
Consistency. The library sleeps on all errors <= to the threshold. When the error is 0, it's internally treated as 1, so it will be slept if 1 <= threshold. If 1 > threshold, it will be raised.